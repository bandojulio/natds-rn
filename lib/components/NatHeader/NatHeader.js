/* eslint-disable */
import React from 'react';
import { Image } from 'react-native';
import PropTypes from 'prop-types';

import { HeaderWrapper, HeaderTitle, HeaderIcon } from './NatHeader.styles';
import { icons } from '../../assets/theme/icons';

export const NatHeader = ({ title, icon }) => {
  return (
    <HeaderWrapper>
      <HeaderIcon>
        <Image source={ icons[icon] } />
      </HeaderIcon>
      <HeaderTitle>{ title }</HeaderTitle>
    </HeaderWrapper>
  )
};

NatHeader.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.string,
};
