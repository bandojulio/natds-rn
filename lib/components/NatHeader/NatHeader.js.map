{"version":3,"sources":["NatHeader.js"],"names":["NatHeader","title","icon","onPress","textRight","iconRight","onPressRight","Error","Platform","OS","AndroidStyle","android","icons","color","fontSize","propTypes","PropTypes","string","func","isRequired","defaultProps"],"mappings":"8IAAA,4B;AACA;;;AAGA,qC;;AAEA;;;;;;;AAOA,+C;;AAEO,GAAMA,6BAAY,QAAZA,UAAY;;AAEnB,IADJC,MACI,MADJA,KACI,CADGC,IACH,MADGA,IACH,CADSC,OACT,MADSA,OACT,CADkBC,SAClB,MADkBA,SAClB,CAD6BC,SAC7B,MAD6BA,SAC7B,CADwCC,YACxC,MADwCA,YACxC;AACJ,GAAI,CAACL,KAAL,CAAY;AACV,KAAM,IAAIM,MAAJ,CAAU,+BAAV,CAAN;AACD;;AAED;AACE,8BAAC,wBAAD;AACE,MAAOC,sBAASC,EAAT,GAAgB,SAAhB,EAA6BC,wBAAaC,OADnD;AAEE,eAFF;AAGE,kBAAkB,QAHpB;;AAKGT;AACD,8BAAC,qBAAD;AACE,QAASC,OADX;AAEE,kBAAkB,QAFpB;;AAIE,8BAAC,kBAAD;AACE,OAAQS,aAAMV,IAAN,CADV;AAEE,eAFF;AAGE,mBAAmB,oDAHrB,iDAJF,CANF;;;;;AAkBE,8BAAC,sBAAD;AACE,KAAMA,IADR;AAEE,sBAAuBD,KAFzB;;AAIGA,KAJH,CAlBF;;AAwBE,8BAAC,sBAAD,EAAa,kBAAkB,QAA/B;AACE,8BAAC,6BAAD,EAAkB,QAASK,YAA3B;AACE,8BAAC,iBAAD,EAAM,MAAO,CAAEO,MAAO,QAAT,CAAmBC,SAAU,EAA7B,CAAb;AACG,GADH;AAEGV,SAFH;AAGG,GAHH,CADF;;AAME,8BAAC,kBAAD;AACE,eADF;AAEE,OAAQQ,aAAMP,SAAN,CAFV;AAGE,mBAAmB,uDAHrB,iDANF,CADF,CAxBF,CADF;;;;;;AAyCD,CAhDM;;AAkDPL,UAAUe,SAAV,CAAsB;AACpBb,KAAMc,oBAAUC,MADI;AAEpBZ,UAAWW,oBAAUC,MAFD;AAGpBd,QAASa,oBAAUE,IAAV,CAAeC,UAHJ;AAIpBb,aAAcU,oBAAUE,IAJJ;AAKpBd,UAAWY,oBAAUC,MALD;AAMpBhB,MAAOe,oBAAUC,MAAV,CAAiBE,UANJ,CAAtB;;;AASAnB,UAAUoB,YAAV,CAAyB;AACvBlB,KAAM,EADiB,CAAzB","file":"NatHeader.js","sourcesContent":["import React from 'react';\nimport {\n  Image, Platform, Text, TouchableOpacity,\n} from 'react-native';\nimport PropTypes from 'prop-types';\n\nimport {\n  HeaderWrapper,\n  HeaderTitle,\n  HeaderIcon,\n  HeaderRight,\n  AndroidStyle,\n} from './NatHeader.styles';\nimport { icons } from '../../assets/theme/icons';\n\nexport const NatHeader = ({\n  title, icon, onPress, textRight, iconRight, onPressRight,\n}) => {\n  if (!title) {\n    throw new Error('Title should not empty string');\n  }\n\n  return (\n    <HeaderWrapper\n      style={Platform.OS === 'android' && AndroidStyle.android}\n      accessible\n      accessibilityRole=\"header\"\n    >\n      {icon && (\n      <HeaderIcon\n        onPress={onPress}\n        accessibilityRole=\"button\"\n      >\n        <Image\n          source={icons[icon]}\n          accessible\n          accessibilityLabel=\"navegação para retornar a página anterior\"\n        />\n      </HeaderIcon>\n      )}\n\n      <HeaderTitle\n        icon={icon}\n        accessibilityLabel={`${title}`}\n      >\n        {title}\n      </HeaderTitle>\n      <HeaderRight accessibilityRole=\"button\">\n        <TouchableOpacity onPress={onPressRight}>\n          <Text style={{ color: 'orange', fontSize: 14 }}>\n            {' '}\n            {textRight}\n            {' '}\n          </Text>\n          <Image\n            accessible\n            source={icons[iconRight]}\n            accessibilityLabel=\"realizar ação do item da barra superior direita\"\n          />\n        </TouchableOpacity>\n      </HeaderRight>\n    </HeaderWrapper>\n  );\n};\n\nNatHeader.propTypes = {\n  icon: PropTypes.string,\n  iconRight: PropTypes.string,\n  onPress: PropTypes.func.isRequired,\n  onPressRight: PropTypes.func,\n  textRight: PropTypes.string,\n  title: PropTypes.string.isRequired,\n};\n\nNatHeader.defaultProps = {\n  icon: '',\n};\n"]}