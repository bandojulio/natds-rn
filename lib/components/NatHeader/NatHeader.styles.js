import styled from 'styled-components/native';
import { colors } from '../../assets/theme/NatColors';

export const HeaderWrapper = styled.View`
  flex-wrap: wrap;
  width: 100%;
  height: 56px;
  background-color: ${colors.primary.white};
`;

export const HeaderIcon = styled.View`
  margin: 18px 0px 18px 18px;
  width: 18px;
`;

export const HeaderTitle = styled.Text`
  margin-left: 20px;
  margin-top: 16px;
  font-size: 18px;
  color: ${colors.bw.grayDarker};
`;
