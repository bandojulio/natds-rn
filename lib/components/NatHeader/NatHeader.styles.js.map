{"version":3,"sources":["NatHeader.styles.js"],"names":["AndroidStyle","StyleSheet","create","android","marginTop","HeaderWrapper","styled","View","colors","primary","white","Platform","select","css","ios","HeaderIcon","TouchableOpacity","HeaderTitle","Text","bw","grayDarker","natura","regular","HeaderRight","yellow"],"mappings":"mwDAAA;AACA,gD;AACA;AACA,qD;;AAEO,GAAMA,mCAAeC,wBAAWC,MAAX,CAAkB;AAC5CC,QAAS;AACPC,UAAW,CADJ,CADmC,CAAlB,CAArB;;;;AAMA,GAAMC,qCAAgBC,iBAAOC,IAAvB;;;;;;;AAOSC,eAAOC,OAAP,CAAeC,KAPxB;AAQTC,sBAASC,MAAT,CAAgB;AAChBT,WAASU,WAAT,mBADgB;;;AAIhBC,OAAKD,WAAL,mBAJgB,CAAhB,CARS,CAAN;;;;;;;AAmBA,GAAME,+BAAaT,iBAAOU,gBAApB;;AAETL,sBAASC,MAAT,CAAgB;AAChBE,OAAKD,WAAL,mBADgB,CAAhB,CAFS,CAAN;;;;;;;AAUA,GAAMI,iCAAcX,iBAAOY,IAArB;;;AAGFV,eAAOW,EAAP,CAAUC,UAHR;AAIIC,cAAOC,OAJX,CAAN;;;AAOA,GAAMC,iCAAcjB,iBAAOC,IAArB;;AAEHC,eAAOC,OAAP,CAAee,MAFZ;AAGTb,sBAASC,MAAT,CAAgB;AAChBT,WAASU,WAAT,mBADgB;;;;;AAMhBC,OAAKD,WAAL,mBANgB,CAAhB,CAHS,CAAN","file":"NatHeader.styles.js","sourcesContent":["import { Platform, StyleSheet } from 'react-native';\nimport styled, { css } from 'styled-components/native';\nimport { colors } from '../../assets/theme/colors/colors';\nimport { natura } from '../../assets/theme/fonts/fonts';\n\nexport const AndroidStyle = StyleSheet.create({\n  android: {\n    marginTop: 4,\n  },\n});\n\nexport const HeaderWrapper = styled.View`\n  flex-wrap: wrap;\n  width: 100%;\n  height: 56;\n  padding-top: 16;\n  padding-bottom: 16;\n  padding-left: 16;\n  background-color: ${colors.primary.white};\n  ${Platform.select({\n    android: css`\n      flex-direction: row;\n    `,\n    ios: css`\n      flex-direction: row;\n      justify-content: space-between;\n    `,\n  })}\n`;\n\nexport const HeaderIcon = styled.TouchableOpacity`\n  width: 24;\n  ${Platform.select({\n    ios: css`\n      flex: 1;\n      justify-content: flex-start;\n    `,\n  })}\n`;\n\nexport const HeaderTitle = styled.Text`\n  margin-left: 24;\n  font-size: 20;\n  color: ${colors.bw.grayDarker};\n  font-family: ${natura.regular};\n`;\n\nexport const HeaderRight = styled.View`\n  width: 60;\n  color:${colors.primary.yellow};\n  ${Platform.select({\n    android: css`\n      flex:1;\n      align-items: flex-end;\n      margin-right:14;\n    `,\n    ios: css`\n      flex: 1;\n      justify-content: flex-end;\n    `,\n  })\n}\n`;\n"]}