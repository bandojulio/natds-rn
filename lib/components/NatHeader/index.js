Object.defineProperty(exports,"__esModule",{value:true});var _NatHeader=require('./NatHeader');Object.defineProperty(exports,'NatHeader',{enumerable:true,get:function get(){return _NatHeader.NatHeader;}});
//# sourceMappingURL=index.js.map