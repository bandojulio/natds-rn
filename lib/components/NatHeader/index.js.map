{"version":3,"sources":["index.js"],"names":["NatHeader"],"mappings":"+LAASA,S","file":"index.js","sourcesContent":["export { NatHeader } from './NatHeader';\n"]}