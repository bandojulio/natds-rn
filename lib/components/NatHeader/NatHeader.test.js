/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';

import { NatHeader } from './NatHeader';

describe('Header component', () => {
  it('Should render container component correctly', () => {
    const header = renderer.create(
      <NatHeader title='Title test'>
      </NatHeader>,
    ).toJSON();

    expect(header).toMatchSnapshot();
  });

  it('Should render header component with title', () => {
    const header = renderer.create(
      <NatHeader title='Title test'>
      </NatHeader>,
    ).toJSON();
    expect(header.children[1].children[0]).toBe('Title test');
  });
});
