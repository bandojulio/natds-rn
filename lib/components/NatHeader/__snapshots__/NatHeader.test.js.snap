// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render container component correctly 1`] = `
<View
  accessibilityRole="header"
  accessible={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexWrap": "wrap",
        "height": 56,
        "width": "100%",
      },
    ]
  }
>
  <Text
    accessibilityLabel="Title test"
    icon=""
    style={
      Array [
        Object {
          "color": "#333333",
          "fontFamily": "NaturaSans-Regular",
          "fontSize": 18,
          "marginLeft": 16,
          "marginTop": 16,
        },
      ]
    }
  >
    Title test
  </Text>
</View>
`;
